.github/workflows/build-release-prod.yml

```diff
@@ -61,17 +68,106 @@ jobs:
             release/**/*.msi
 
       - uses: actions/upload-artifact@v2
-        name: Upload MacOS build
+        name: macos-build
         if: matrix.os == 'macos-latest'
         with:
           name: stacks-wallet-prod-${{ steps.vars.outputs.version }}-macos
           path: release/**/*.dmg
 
       - uses: actions/upload-artifact@v2
-        name: Upload Linux build
-        if: matrix.os == 'ubuntu-latest'
+        name: linux-build
+        if: matrix.os == 'ubuntu-20.04'
         with:
           name: stacks-wallet-prod-${{ steps.vars.outputs.version }}-linux
           path: |
             release/**/*.deb
             release/**/*.rpm
+
+  release:
+    runs-on: ubuntu-20.04
+
+    needs: build
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Run ID ${{ github.run_id }}
+        run: echo "${{ github.run_id }}"
+
+      - name: Semantic Release
+        uses: cycjimmy/semantic-release-action@v2.5.0
+        id: semantic
+        if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN || secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          SEMANTIC_RELEASE_PACKAGE: ${{ github.workflow }}
+        with:
+          # These plugins aren't packaged with semantic-release by default. So specify them here to ensure they get installed during this GH Action
+          extra_plugins: |
+            @semantic-release/changelog
+            @semantic-release/git
+            semantic-release-slack-bot
+            @semantic-release/exec
+
+      - uses: actions/download-artifact@v2
+        with:
+          path: release/
+
+      - run: ls -la release
+
+      # - uses: dawidd6/action-download-artifact@v2
+      #   with:
+      #     # Download all artifacts from this job
+      #     workflow: build-release-prod.yml
+      #     run_id: ${{ github.run_id }}
+      #     # workflow: build-release-prod.yml
+      #     path: release
+
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: v${{ steps.semantic.outputs.release-version }}
+          release_name: Release ${{ steps.semantic.outputs.release-version }}
+          draft: true
+          prerelease: true
+
+      - name: List files
+        run: ls -R .
+
+      - name: Upload Windows build
+        id: upload-win-asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
```

Comment:
@CharlieC3 [following up on your comment on devops](https://github.com/blockstackpbc/devops/issues/516#issuecomment-740006680), because we need to later use `actions/upload-release-asset@v1` to upload wallet assets we need this output from `create_release`, I'm not sure how we can use the semantic-release plugin as it doesn't output the release id (afait).

Don't we need to use both, creating the release with the version output of `https://github.com/cycjimmy/semantic-release-action#outputs` ?

Reply:
The semantic release _github_ plugin can be configured in the `package.json` to upload any defined distribution assets. [See full config info here](https://github.com/semantic-release/github#usage), but generally you would need to add something like the following to your `package.json`:
```
...
    ["@semantic-release/github", {
      "assets": [
        {"path": "dist/*.msi", "label": "Windows distribution"},
        {"path": "dist/*.dmg", "label": "OSX distribution"},
        {"path": "dist/*.rpm", "label": "Linux distribution"},
        {"path": "dist/*.deb", "label": "Debian distribution"}
      ]
    }],
...
```

You'd still need to download the artifacts to the `dist` working directory before the semantic release action run, [just like you're already doing](https://github.com/blockstack/stacks-wallet/blob/fix/update-ci/.github/workflows/build-release-prod.yml#L121) before uploading the release artifacts.